modules/backend/widgets/MediaManager.php

```diff
@@ -109,6 +110,32 @@ public function render(): string
     // AJAX handlers
     //
 
+    /**
+     * Process file uploads submitted via AJAX
+     *
+     * @throws ApplicationException If the file "file_data" wasn't detected in the request or if the file failed to pass validation / security checks
+     */
+    public function onUpload(): ?\Illuminate\Http\Response
+    {
+        ini_set('memory_limit', Config::get('cms.storage.media.memoryLimit', ini_get('memory_limit')));
```

Comment:
I had explained earlier when i was uploading large files I got a memory exhausted error (this was after fixing the error above). I can move it out if it works in the event listener (at the same time i was unsure if the event listener even work because everything else uses `$this->fireSystemEvent` that I thought that Event was unused.

Reply:
A memory exhausted error feels more like the image is being processed in some way, I don't think I've seen it happen just from a file being uploaded before. I'd need more evidence that this is actually a problem that affects most Winter users before I'd want it in the core, you should be able to implement for your own project just with the existing event listener for now though.